src/lightning/pytorch/utilities/parsing.py

```diff
@@ -204,7 +205,7 @@ def save_hyperparameters(
     obj._hparams_initial = copy.deepcopy(obj._hparams)
 
 
-class AttributeDict(Dict):
+class AttributeDict(_AttributeDict):
```

Comment:
Since the name is the same, we can now simply import the fabric one? Or did you keep the subclass for a reason?

Reply:
I did because the docstring is very different.